pkg/dclient/client.go

```diff
@@ -29,6 +29,7 @@ type Client struct {
 	client          dynamic.Interface
 	log             logr.Logger
 	clientConfig    *rest.Config
+	Clientset       *kubernetes.Clientset
```

Comment:
should be private.

Reply:
Obtaining api data through raw ptah requires Clientset, which cannot be private.

Reply:
Ah right, should not be added there then. Why no in `PolicyContext` instead ?

Reply:
I was thinking, it's simpler in `Client`.

Reply:
`dclient` is our custom dynamic client, it only needs a `dynamic.Interface`. I see no reason to add it there.

Reply:
[Here](https://github.com/kyverno/kyverno/blob/96b33f62004630ab974e4ec7902a152aefe76bd3/pkg/engine/policyContext.go#L29) would be the correct place IMHO.

Reply:
I don't think we even need to reconstruct the kclient separately. We should be able to add a method to the Client similar to GetResource which can fetch resources given a raw api resource endpoint. Something like GetResourceRaw which uses the existing private kclient.